src/Umbraco.Web.UI.Client/src/common/directives/validation/valservermatch.directive.js

```diff
@@ -0,0 +1,89 @@
+
+function valServerMatch(serverValidationManager) {
+    return {
+        require: ['form', '^^umbProperty', '?^^umbVariantContent'],
+        restrict: "A",
+        scope: {
+            valServerMatch: "="
+        },
+        link: function (scope, element, attr, ctrls) {
+
+            var formCtrl = ctrls[0];
+            var umbPropCtrl = ctrls[1];
+            if (!umbPropCtrl) {
+                //we cannot proceed, this validator will be disabled
+                return;
+            }
+
+            // optional reference to the varaint-content-controller, needed to avoid validation when the field is invariant on non-default languages.
+            var umbVariantCtrl = ctrls[2];
+
+            var currentProperty = umbPropCtrl.property;
+            var currentCulture = currentProperty.culture;
+            var currentSegment = currentProperty.segment;
+
+            if (umbVariantCtrl) {
+                //if we are inside of an umbVariantContent directive
+
+                var currentVariant = umbVariantCtrl.editor.content;
+
+                // Lets check if we have variants and we are on the default language then ...
+                if (umbVariantCtrl.content.variants.length > 1 && (!currentVariant.language || !currentVariant.language.isDefault) && !currentCulture && !currentSegment && !currentProperty.unlockInvariantValue) {
+                    //This property is locked cause its a invariant property shown on a non-default language.
+                    //Therefor do not validate this field.
+                    return;
+                }
+            }
+
+            // if we have reached this part, and there is no culture, then lets fallback to invariant. To get the validation feedback for invariant language.
+            currentCulture = currentCulture || "invariant";
+            
+            var unsubscribe = [];
+
+            //subscribe to the server validation changes
+            function serverValidationManagerCallback(isValid, propertyErrors, allErrors) {
+                if (!isValid) {
+                    formCtrl.$setValidity('valServerMatch', false);                    
+                }
+                else {
+                    formCtrl.$setValidity('valServerMatch', true);                    
```

Comment:
@nielslyngsoe yeah not sure why our .editorconfig isn't kicking in here.... oh, i just looked it's because this is the setting: 🤦‍♂️ 

```
[*.{js,less}]
trim_trailing_whitespace = false
```

I have no idea why this is set to false do you? Seems like it should absolutely be set to true?

Reply:
I'm going to push a change to keep the whitespace for .less but remove it for .js (Since I'm not sure what requirements there are for .less). Let me know if you want to change that.